daemon/oci_linux.go

```diff
@@ -98,6 +98,30 @@ func setDevices(s *specs.Spec, c *container.Container) error {
 		for _, d := range hostDevices {
 			devs = append(devs, oci.Device(d))
 		}
+		for _, deviceMapping := range c.HostConfig.Devices {
+			if deviceMapping.PathOnHost == deviceMapping.PathInContainer {
+				logrus.Warnf("Specifying a device %s is redundant for privileged mode", deviceMapping.PathOnHost)
+				if deviceMapping.CgroupPermissions != "rwm" {
+					logrus.Warnf("Specify  permission %s to device %s   is redundant for privileged mode", deviceMapping.CgroupPermissions, deviceMapping.PathOnHost)
+				}
+				continue
+			}
+			if _, err := os.Stat(deviceMapping.PathInContainer); os.IsNotExist(err) {
+				if deviceMapping.CgroupPermissions != "rwm" {
+					logrus.Warnf("Specify  permission %s to device %s is redundant for privileged mode", deviceMapping.CgroupPermissions, deviceMapping.PathOnHost)
+				}
+				d, _, err := oci.DevicesFromPath(deviceMapping.PathOnHost, deviceMapping.PathInContainer, "rwm")
+				if err != nil {
+					return err
+				}
+				devs = append(devs, d...)
+			} else {
+				if err == nil {
+					return fmt.Errorf("In privileged mode,Path:%s in Container should be different with any path on host ", deviceMapping.PathInContainer)
```

Comment:
`return errdefs.InvalidParameter(errors.New(errorMsg))`

Additionally this should be validated on container create.